test/crit/main.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit"
+)
+
+func main() {
+	imgs := getImgs()
+	for _, img := range imgs {
+		log.Println("=== ", img)
+		testImg := fmt.Sprintf("%s.test.img", filepath.Base(img))
+		c := crit.New(img, testImg, "", false, false)
+		// Decode the binary image file
+		decodedImg, err := c.Decode()
+		if err != nil {
+			log.Fatal("[DECODE]: ", err)
+			continue
+		}
+		// Encode it into test binary image file
+		err = c.Encode(decodedImg)
+		if err != nil {
+			log.Fatal("[ENCODE]: ", err)
+			continue
+		}
+		// Open and compare original and test files
+		imgBytes, err := os.ReadFile(img)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		testImgBytes, err := os.ReadFile(testImg)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+		if !bytes.Equal(imgBytes, testImgBytes) {
+			log.Fatal("[RECODE]: files do not match")
+		}
+		log.Println("=== PASS")
+	}
+}
+
+func getImgs() []string {
+	var skipImgs = []string{
+		"pages-",
+		"iptables-",
+		"ip6tables-",
+		"nftables-",
+		"route-",
+		"route6-",
+		"ifaddr-",
+		"tmpfs-",
+		"netns-ct-",
+		"netns-exp-",
+		"rule-",
+	}
+	var imgs []string
+	dir, err := os.ReadDir(".")
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range dir {
+		name := file.Name()
+		if filepath.Ext(name) == ".img" {
+			for _, skip := range skipImgs {
+				if strings.HasPrefix(name, skip) {
+					goto skipFile
```

Comment:
can we use `break` instead of `goto skipFile`?

Reply:
`break` will get us out of the first for loop (which checks against the array of excluded patterns) but will still end up appending the image to the list in the outer for loop. Our options are either a flag or a label, and this one seemed cleaner. If you would like, I can change it to use a `bool skipFile = true` instead.

Reply:
Both [`break`](https://go.dev/doc/go1.17_spec#Break_statements) and [`continue`](https://go.dev/doc/go1.17_spec#Continue_statements) support labels but perhaps `continue` makes more sense in the current context:
```diff
--- a/test/crit/main.go
+++ b/test/crit/main.go
@@ -66,17 +66,18 @@ func getImgs() []string {
        if err != nil {
                log.Fatal(err)
        }
+
+nextFile:
        for _, file := range dir {
                name := file.Name()
                if filepath.Ext(name) == ".img" {
                        for _, skip := range skipImgs {
                                if strings.HasPrefix(name, skip) {
-                                       goto skipFile
+                                       continue nextFile
                                }
                        }
                        imgs = append(imgs, file.Name())
                }
-       skipFile:
        }
 
        return imgs
```